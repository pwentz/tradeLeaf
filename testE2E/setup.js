import React, { Component } from 'react';
const { View, StyleSheet } = require('react-native');

import { Provider } from 'react-redux';

import MockApi from './mocks/MockApi';
const { setupApp } = require('../setupApp');

__DEV__ = true;

jest.mock('Linking', () => {
  return {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    openURL: jest.genMockFn().mockReturnValue(Promise.resolve()),
    canOpenURL: jest.genMockFn().mockReturnValue(Promise.resolve()),
    getInitialURL: jest.genMockFn().mockReturnValue(Promise.resolve()),
  };
});

jest.mock('ScrollView', () => jest.genMockFromModule('ScrollView'));

export const renderer = require('react-test-renderer');

const AccountRequirements = require('../src/components/register/AccountRequirements').default;
const App = require('../src/components/app/App').default;
const Avatar = require('../src/components/common/Avatar').default;
const Card = require('../src/components/matchboard/Card').default;
const LoginForm = require('../src/components/login/LoginForm').default;
const RegisterForm = require('../src/components/register/RegisterForm').default;
const TabBar = require('../src/components/app/TabBar').default;
const ChatPreview = require('../src/components/chat/ChatPreview').default;

export const components = {
  LoginForm,
  RegisterForm,
  Card,
  AccountRequirements,
  Avatar,
  TabBar,
  ChatPreview,
};

class TestRouteObserver {}

export function setupTestApp() {
  const api = new MockApi();

  const shouldLog = false;

  const { actions, store } = setupApp(api, shouldLog);

  const styles = StyleSheet.create({
    container: {
      flex: 1,
      justifyContent: 'center',
      alignItems: 'stretch',
      backgroundColor: 'white',
    },
  });

  const testRouteObserver = new TestRouteObserver();

  class TradeLeafApp extends Component {
    render() {
      return (
        <View style={styles.container}>
          <Provider store={store}>
            <App screenProps={{ actions, testRouteObserver }} />
          </Provider>
        </View>
      );
    }
  }

  return {
    TradeLeafApp,
    api,
    actions,
    dispatch: store.dispatch,
    testRouteObserver,
  };
}
