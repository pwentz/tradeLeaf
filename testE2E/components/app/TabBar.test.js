import React from 'react';

import { components, renderer } from '../../setup';

const noop = () => {}
const routes = ['Home', 'Search', 'Notifications', 'Inbox']
const nav = { navigate: noop, state: { routes, index: 0 } }

it('<TabBar />', () => {
  const TabBar = components.TabBar;
  const component = renderer.create(
    <TabBar
      navigation={nav}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
