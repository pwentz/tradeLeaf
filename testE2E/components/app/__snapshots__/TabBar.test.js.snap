// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabBar /> 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
    }
  }
>
  <View
    style={
      Object {
        "height": 200.1,
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginLeft": "3%",
          "marginTop": "7.5%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "borderRadius": 25,
              "height": 50,
              "width": 50,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/images/tradeLeafIcon.png",
            }
          }
          style={
            Array [
              Object {
                "borderColor": "#B8B8B8",
                "borderWidth": 1,
              },
              Object {
                "borderRadius": 25,
                "height": 50,
                "width": 50,
              },
            ]
          }
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#DFBA00",
            "fontSize": 20,
            "marginBottom": "1%",
            "marginLeft": "5%",
          }
        }
      >
        home
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#B8B8B8",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "width": 750,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#DFBA00",
          "borderBottomWidth": 4,
          "flex": 1,
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#DFBA00",
              "fontSize": 28,
            },
            undefined,
            Object {
              "fontFamily": "Octicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#B8B8B8",
          "borderBottomWidth": 1,
          "flex": 1,
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#B8B8B8",
              "fontSize": 28,
            },
            undefined,
            Object {
              "fontFamily": "Octicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#B8B8B8",
          "borderBottomWidth": 1,
          "flex": 1,
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#B8B8B8",
              "fontSize": 28,
            },
            undefined,
            Object {
              "fontFamily": "Octicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#B8B8B8",
          "borderBottomWidth": 1,
          "flex": 1,
          "height": 48,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#B8B8B8",
              "fontSize": 28,
            },
            undefined,
            Object {
              "fontFamily": "Octicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
